pkg/filters/corsadaptor/corsadaptor.go

```diff
@@ -120,24 +121,28 @@ func (a *CORSAdaptor) Handle(ctx *context.Context) string {
 
 func (a *CORSAdaptor) handle(ctx *context.Context) string {
 	r := ctx.Request().(*httpprot.Request)
-	w := ctx.Response().(*httpprot.Response)
 	method := r.Method()
 	headerAllowMethod := r.Header().Get("Access-Control-Request-Method")
 	if method == http.MethodOptions && headerAllowMethod != "" {
-		a.cors.HandlerFunc(w.Std(), r.Std())
+		rw := httptest.NewRecorder()
```

Comment:
I think this is ok, but would be better if we can avoid using `httptest`, as it it for testing.